connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceConnectorTest.java

```diff
@@ -319,4 +320,17 @@ public void testRefreshTopicPartitionsTopicOnTargetFirst() throws Exception {
         connector.refreshTopicPartitions();
         verify(connector, times(1)).computeAndCreateTopicPartitions();
     }
+
+    @Test
+    public void testIsCycleWithNullUpstreamTopic() throws Exception {
+        class BadReplicationPolicy extends DefaultReplicationPolicy {
```

Comment:
What about `CustomReplicationPolicy` as it's not really a bad policy, `null` is allowed

Reply:
> What about `CustomReplicationPolicy` as it's not really a bad policy, `null` is allowed

`CustomReplicationPolicy` will not fail the test since it ends up return `false` after the first line of `isCycle` returns null:
```java
boolean isCycle(String topic) {
    String source = replicationPolicy.topicSource(topic);
    //...
```

Reply:
I just meant renaming `BadReplicationPolicy` into `CustomReplicationPolicy`

Reply:
Ah sorry, I misunderstood. I just now renamed it as suggested.